packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -775,21 +788,34 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
         }
       }, [])
 
-      return useMemo(
-        () => ({
-          /**
-           * A method to manually refetch data for the query
-           */
-          refetch: () => {
-            if (!promiseRef.current)
-              throw new Error(
-                'Cannot refetch a query that has not been started yet.'
-              )
-            return promiseRef.current?.refetch()
+      return {
+        /**
+         * A method to manually refetch data for the query
+         */
+        refetch: useCallback(() => {
+          if (!promiseRef.current)
+            throw new Error(
+              'Cannot refetch a query that has not been started yet.'
+            )
+          return promiseRef.current?.refetch()
+        }, []),
+        /**
+         * a `.then` method that allows the return value to be used with the `use` hook like
+         * `const result = use(useSomeQuery(args))`
+         */
+        then: useCallback<Promise<unknown>['then']>(
+          (onfulfilled, onrejected) => {
+            initiateQueryIfNeeded()
```

Comment:
The question is still open if React will throw away the current component state & refs when suspending before the component was rendered at least once.

If it would, this would not work and we would have to try something with `getRunningQueryThunk` here instead. But let's hope that we won't have that problem in the future.